js/components/chart-base.tsx

```diff
@@ -165,7 +165,9 @@ class ChartBase extends React.Component<IChartBaseProps, null> {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.isLoading && !nextProps.isLoading)
+        const isEqual = require('react-fast-compare');
+
+        if ((this.props.isLoading && !nextProps.isLoading) || isEqual(this.props.objectData, nextProps.objectData))
             setTimeout(() => this.updateChart());
```

Comment:
I know this was here before, but could you find an alternative to using setTimeout?
It seems to me like it shouldn't be needed.